andela/ah-backend-stark
# #160584003 users should be able to follow each other

#### What does this PR do?
`allows users to follow each other and to manage people they are following`
#### Description of Task to be completed?
Have the following endpoints working:
`POST api/profile/<str:username>/follow/`
`DELETE api/profile/<str:username>/unfollow/`
`GET api/profile/<str:username>/following/`
`GET api/profile/<str:username>/followers/`
#### How should this be manually tested?
After pulling the changes from this branch migrations should be made using:
```
 - $ python manage.py makemigrations articles
 - $ python manage.py makemigrations authentication
 - $ python manage.py makemigrations profiles
 - $ python manage.py migrate 
```

Using Postman to run the application, you can run the endpoints above to:
```
- follow users
- unfollow users
- look at your followers
- look at other users' followers
- look at users you are following
- look at users other users are following
```
#### What are the relevant pivotal tracker stories?
#160584003 - Users should be able to follow each other
#### Screenshots (if appropriate)
Get users who you are following
![image](https://user-images.githubusercontent.com/40589388/47015794-9f9c1880-d156-11e8-870c-b60ffece45be.png)

Get followers
![image](https://user-images.githubusercontent.com/40589388/47015836-c35f5e80-d156-11e8-99dd-8975bc40114a.png)

Message after following "tester4"
![image](https://user-images.githubusercontent.com/40589388/47015913-f4d82a00-d156-11e8-8e2c-d56ace7ad6e5.png)

Message after unfollowing "tester4"
![image](https://user-images.githubusercontent.com/40589388/47015955-0faa9e80-d157-11e8-908e-fd4ba207dd0f.png)

#### Questions:
